java/daikon/dcomp/BuildJDK.java

```diff
@@ -3,375 +3,500 @@
         options.printUsage();
-        return false;
+        System.exit(1);
       }
     } else {
-      if (target_args.length < 2) {
-        System.out.println("must specify source jar and destination dir");
+      if (target_args.length < 1) {
+        System.out.println("must specify destination dir");
         options.printUsage();
-        return false;
+        System.exit(1);
       }
-      if (target_args.length > 3) {
-        System.out.println("too many arguments");
+      if (target_args.length > 1) {
+        System.out.println("too many arguments: found " + target_args.length + ", expected 1");
         options.printUsage();
-        return false;
+        System.exit(1);
       }
     }
 
-    return true;
-  }
+    String JAVA_HOME = System.getenv("JAVA_HOME");
+    if (JAVA_HOME == null) {
+      if (verbose) {
+        System.out.println("JAVA_HOME not defined; using java.home: " + java_home);
+      }
+      JAVA_HOME = java_home;
+    }
 
-  private static JarFile getJarFile(String potentialJarFileName) throws IOException, ZipException {
-    JarFile jfile;
-    final String p = potentialJarFileName;
-    try {
-      String jar_name = findRtJarFilename(p);
-      System.out.printf("using jar file %s%n", jar_name);
-      jfile = new JarFile(jar_name);
-    } catch (ZipException e) {
-      throw new ZipException(e.getMessage() + "; filename was " + p);
+    File jrt = new File(JAVA_HOME);
+    if (!jrt.exists()) {
+      System.out.printf("Java home directory %s does not exist.%n", jrt);
+      System.exit(1);
     }
-    return jfile;
-  }
 
-  private static String findRtJarFilename(final String arg) {
-    return new RtJarFinder(arg).findRtJar("/lib/rt.jar /jre/lib/rt.jar");
-  }
+    try {
+      jrt = jrt.getCanonicalFile();
+    } catch (Exception e) {
+      System.out.printf("Error geting canonical file for %s: %s", jrt, e.getMessage());
+      System.exit(1);
+    }
 
-  void copyStreams(InputStream fis, OutputStream fos) throws Exception {
-    byte[] buf = new byte[1024];
-    int i = 0;
-    while ((i = fis.read(buf)) != -1) {
-      fos.write(buf, 0, i);
+    JAVA_HOME = jrt.getAbsolutePath();
+    if (!java_home.startsWith(JAVA_HOME)) {
+      System.out.printf(
+          "JAVA_HOME (%s) does not agree with java.home (%s).%n", JAVA_HOME, java_home);
+      System.out.printf("Please correct your Java environment.%n");
+      System.exit(1);
     }
-    fis.close();
-    fos.close();
   }
 
-  void translate_classes(JarFile jfile, String dest, String prefix, String prefixOfFilesToInclude)
-      throws java.io.IOException {
-
-    // Map from classname to class so we can find out information about
-    // classes we have not yet instrumented.
-    Map<String, JavaClass> classmap = new LinkedHashMap<>();
+  /**
+   * For Java 8 the Java runtime is located in rt.jar. This method creates an InputStream for each
+   * class in rt.jar and saves this information in {@link #class_stream_map}.
```

Comment:
I don't think there is any need for a `class_stream_map` field.
Instead, this method could create and return the map, and the client (that is, `main`) could pass the map to `instrument_classes`.  That would clarify exactly where and how the map is used, and eliminate an unneeded field.

Reply:
I went ahead and made these changes - and perhaps I did not do it the way you expected, but I found the results to be a bit awkward with several more lines of code and extra arguments.